lib/puppet/provider/kernel_parameter/grub.rb

```diff
@@ -13,6 +13,7 @@
   lens { 'Grub.lns' }
 
   confine :feature => :augeas
+  confine :grubversion => 1
```

Comment:
why not use defaultfor? maybe there are people around that wan't to the old grub or the other way around. but if you confine the version you can't set the provider on the resource with "provider => grub2"